### DISCUSSION TITLE
Dedicated link body syntax

### DISCUSSION THREAD

**Discussion by dryya** [typst/typst#1667] [Ideas]
I really appreciate that Typst includes a lot of markdown-inspired dedicated syntax. Would shorthand for the `#link` function be in scope for this project?
Possibilities include:

- Asciidoc style: `https://asciidoctor.org[Asciidoctor]` Since there is already a dedicated syntax to recognize `https://` (and maybe other URI schemes? I'm not certain), this seems to make the most sense to me.
- CommonMark style: `[Asciidoctor](https://asciidoctor.org)`. For better or for worse, this is most similar to Typst's function syntax: `#link("https://example.com")[See example.com]`
- Dokuwiki (and many other wikis) style: `[[https://www.example.com|Link to example]]`
- Org mode style: `[[https://www.example.com][Some link]]`
- reStructuredText style (probably not a good fit, sigils are already in use):
```
`Link text <https://www.example.com/>`_
```

**Comment by laurmaedje** [comment]
Can't say that I _love_ the Asciidoctor style syntax, but it would be pretty consistent with the rest of Typst (since it's a trailing content block, like for a function call). The question is just whether links with explicit body are needed often enough to warrant this extra syntax.

**Reply by lcnbr** [reply]
I definitely think this will become much more used when HTML output is done!

**Reply by bfeitknecht** [reply]
I second this. Links (internal but especially external ones) are what make Markdown's usage for instance in [Obsidian](obsidian.md) so attractive. HTML is canonically all about linking pages so when HTML support is finally here, that's what I'm most excited about. I'd love to have the power of linking files and content that Obsidian brings but with Typst as source code file type. With that in mind my suggestion is to stick to the syntax that is already prevalent and in common knowledge `[alias](test.com)`.

**Comment by Eloitor** [comment]
I'm interested in links to other files in the same directory. In markdown the syntax for this is `[link](./file.md#section-name)`. This could be useful both for pdf output and html...

I don't know how could the Asciidoc style fit this usecase
